src/Components/Components/src/PersistentState/PersistentComponentRegistration.cs

```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+namespace Microsoft.AspNetCore.Components;
+
+[DebuggerDisplay($"{{{nameof(GetDebuggerDisplay)}(),nq}}")]
+internal sealed class PersistentComponentRegistration<TService>(IComponentRenderMode componentRenderMode) : IPersistentComponentRegistration
```

Comment:
Should `TService` really be `TComponent`? I'm confused by the naming because the type name seems to imply this is for a component, and the generic param seems to imply it's for a service.